build/xtask/src/dist.rs

```diff
@@ -446,13 +446,16 @@ pub fn package(
 
         translate_srec_to_other_formats(&cfg, image_name, "combined")?;
```

Comment:
I don't believe we can outright drop translate_sec* until we have a better story for the signed code path - I think that would put us on a path to interesting clashes with upstream that I don't think we want.

A follow-up PR will have all formats initially generated from the elf - but retaining translate_srec for the signing path. I'm still verifying objcopy is generating correct images - there's an appreciable size difference in ihex & srec.

Reply:
Correct - it only converts, but in two different flows. In the unsigned image flow, `translate_srec...` is only used to derive the alternative image formats - but in the signed image flow (line 467 onwards), Hubris signs the derived binary, creating a `combined_rsa.bin`, which is then re-converted _back_ to srec (as `final.srec`) before yet _another_ set of derived images (`final.x`) is then derived from that signed srec.

I'm not sure we want to make a call on image signing as yet - so my plan was to leave `translate_srec...` and `write_srec` in place, until we can decide what to do about image signing.

Reply:
Understood about the signing flow. My point was that the common flow can be updated to do:
1. Generate elf
2. Translate elf to other formats

If signing is required, then it would do the `binary_to_srec` and `translate_srec_to_other_formats`. That way you won't need `write_srec` at all and the srec handling would be limited to only the signing path which as you mentioned we don't really want to care about right now. That way, we won't have any special srec generation code in our common path (just less things to worry about from image management standpoint).

Anyways, if you want to retain the srec path, I think it would be good to at least add a comment in the file regarding why it is left around and probably a JIRA issue to track this as a TODO for later.

Reply:
Ah, so there's a bit more to unpack here:

1. This MR cannot derive other formats from the elf. objcopy requires section headers in addition to program headers, even though QEMU is happy enough without.
2. In the patch that's immediately after this, the derived srec is not identical, and I don't yet have a mechanism to confirm if that deviation is material. While it matches in terms of overall size & record count, the records themselves have different header data. Comparing the derived files suggests that objcopy is reordering blocks.